pilot/pkg/networking/core/v1alpha3/listener.go

```diff
@@ -139,16 +137,106 @@ const (
+	//  FCM 4: ALPN [] Transport protocol: N/A                                              --> TCP traffic over plain text
+	//  FCM 5: ALPN [] Transport protocol: tls                                              --> TCP traffic over TLS
+	// If traffic is over plain text or mTLS is strict mode, there are two filter chains. The filter chain match should be
+	//  FCM 1: ALPN [http/1.0, http/1.1, h2c, istio-http/1.0, istio-http/1.1, istio-h2]     --> HTTP traffic over plain text or TLS
+	//  FCM 2: ALPN []                                                                      --> TCP traffic over plain text or TLS
+	inboundPermissiveFilterChainMatchOptions = []FilterChainMatchOptions{
+		{
+			// client side traffic was detected as HTTP by the outbound listener, sent over mTLS
+			ApplicationProtocols: mtlsHTTPALPNs,
+			// If client sends mTLS traffic, transport protocol will be set by the TLS inspector
+			TransportProtocol: "tls",
+			Protocol:          plugin.ListenerProtocolHTTP,
+		},
+		{
+			// client side traffic was detected as HTTP by the outbound listener, sent out as plain text
+			ApplicationProtocols: plaintextHTTPALPNs,
+			// No transport protocol match as this filter chain (+match) will be used for plain text connections
+			Protocol: plugin.ListenerProtocolHTTP,
+		},
+		{
+			// client side traffic could not be identified by the outbound listener, but sent over mTLS
+			ApplicationProtocols: mtlsTCPALPNs,
+			// If client sends mTLS traffic, transport protocol will be set by the TLS inspector
+			TransportProtocol: "tls",
+			Protocol:          plugin.ListenerProtocolTCP,
+		},
+		{
+			// client side traffic could not be identified by the outbound listener, sent over plaintext
+			// or it could be that the client has no sidecar. In this case, this filter chain is simply
+			// receiving plaintext TCP traffic.
+			Protocol: plugin.ListenerProtocolTCP,
+		},
+		{
+			// client side traffic could not be identified by the outbound listener, sent over one-way
+			// TLS (HTTPS for example) by the downstream application.
+			// or it could be that the client has no sidecar, and it is directly making a HTTPS connection to
+			// this sidecar. In this case, this filter chain is receiving plaintext one-way TLS traffic. The TLS
+			// inspector would detect this as TLS traffic [not necessarily mTLS]. But since there is no ALPN to match,
+			// this filter chain match will treat the traffic as just another TCP proxy.
+			TransportProtocol: "tls",
+			Protocol:          plugin.ListenerProtocolTCP,
+		},
+	}
+
+	inboundStrictOrNoneFilterChainMatchOptions = []FilterChainMatchOptions{
+		{
+			// client side traffic was detected as HTTP by the outbound listener.
+			// If we are in strict mode, we will get mTLS HTTP ALPNS only. If we are in plaintext mode,
+			// we would get plaintext HTTP ALPNS only. These two are mutually exclusive. So it is
+			// okay to put both of them into one match condition because we have to generate the same
```

Comment:
There is a TODO item below to move the "strict" to the autoN plugin code so that listener code doesn't need't need to handle the ALPN for "strict" mTLS.

Reply:
Ping.

Reply:
I will send another commit to update this later.

Reply:
Done